module/VuFindSearch/src/VuFindSearch/Backend/EDS/SearchRequestModel.php

```diff
@@ -211,6 +211,27 @@ public function setParameters($parameters = [])
         }
     }
 
+    /**
+     * Checks whether the search model is valid for the EDS API.
+     *
+     * @return bool Returns false if any model parameters are invalid.
+     */
+    public function isValid()
+    {
+        $contentProviderValues = $this->facetFilters['ContentProvider'] ?? [];
+        foreach ($contentProviderValues as $value) {
+            // Content providers should be only alphanumeric+space strings.
+            // This explode + ctype logic is about 8x faster than using a regex.
+            $words = explode(' ', $value);
+            foreach ($words as $word) {
+                if (!ctype_alnum($word)) {
```

Comment:
Argh, no this doesn't work, there are more available characters than alphanumeric.  I will try switching it to check for invalid characters...

Reply:
Replaced this logic to search specifically *for* the invalid `|` character.  I don't see this in any DB we license, and I would be surprised if it shows up in any other, but will ask EBSCO.